main.c

```diff
@@ -652,13 +683,50 @@ static void kbd_leave(void *data, struct wl_keyboard *k, uint32_t serial,
 	reset_repeat();
 }
 
+static xkb_keysym_t process_key(xkb_keysym_t sym)
+{
+	if (!term.xkb_compose_state)
+		return sym;
+	if (sym == XKB_KEY_NoSymbol)
+		return sym;
+	if (xkb_compose_state_feed(term.xkb_compose_state,
+				   sym) != XKB_COMPOSE_FEED_ACCEPTED)
+		return sym;
+
+	switch (xkb_compose_state_get_status(term.xkb_compose_state)) {
+	case XKB_COMPOSE_COMPOSING:
+		return XKB_KEY_NoSymbol;
+	case XKB_COMPOSE_COMPOSED:
+		return xkb_compose_state_get_one_sym(term.xkb_compose_state);
+	case XKB_COMPOSE_CANCELLED:
+		return XKB_KEY_NoSymbol;
+	case XKB_COMPOSE_NOTHING:
+	default:
+		return sym;
+	}
+}
+
 static void kbd_key(void *data, struct wl_keyboard *k, uint32_t serial,
 		    uint32_t time, uint32_t key, uint32_t state)
 {
 	xkb_keysym_t sym;
+	const xkb_keysym_t *syms;
+	uint32_t num_syms;
 	uint32_t unicode;
 
-	sym = xkb_state_key_get_one_sym(term.xkb_state, key + 8);
+	if (state == WL_KEYBOARD_KEY_STATE_RELEASED) {
+		if (term.repeat.key == key)
+			reset_repeat();
+		return;
+	}
+
+	num_syms = xkb_state_key_get_syms(term.xkb_state, key + 8, &syms); 
```

Comment:
why did you replace `xkb_state_key_get_one_sym`? Does this behave different?

Reply:
I thought it would be useful on some key combination to a compose key. However, I never crossed this case. So, the function seems to be "sub used", also keeping in mind that only the first sym is used. `xkb_state_key_get_one_sym` is enough and fits well. I will keep it then.